src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -219,7 +219,7 @@ private static bool TryGetByteArrayFromFileStream(Stream stream, out byte[] buff
             // than the buffer size. The default buffer size is 4KB, so this will incur a 4KB
             // allocation for any files less than 4KB. That's why, for example, the command
             // line compiler actually specifies a very small buffer size.
-            return stream.Read(buffer, 0, length) == length;
+            return stream.TryReadAll(buffer, 0, length) == length;
```

Comment:
It looks like `SourceText.From` explicitly accounts for this case down the line and does the read inside the `Decode`.

My impression is that this was intentional and it is possible to get to this path from `FileStream`. See [`SourceText.Decode`](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/Text/SourceText.cs#L187):

```
// Note: We are setting the buffer size to 4KB instead of the default 1KB. That's
            // because we can reach this code path for FileStreams and, to avoid FileStream
            // buffer allocations for small files, we may intentionally be using a FileStream
            // with a very small (1 byte) buffer. Using 4KB here matches the default buffer
            // size for FileStream and means we'll still be doing file I/O in 4KB chunks.
```